### ISSUE TITLE
PREMERE: Meta-Reweighting via Self-Ensembling for Point-of-Interest Recommendation

### ISSUE THREAD

**Issue by hello-im-yj** [One-Day-One-Paper/One-Day-One-Paper#208]
# PREMERE: Meta-Reweighting via Self-Ensembling for Point-of-Interest Recommendation

🔑 keyword

- ***POI***
    
    소비자가 흥미롭거나 유용하다고 여기는 특정한 장소를 의미한다.
    
- **Sample reweighting** (Ma et al. 2018; Hu, Koren, and Volinsky 2008; Yu, Bilenko, and Lin 2017; Song et al. 2020b)
    
    : increases the learning weight of positive (non-zero) samples which are very valuable owing to their scarcity (Hu, Koren, and Volinsky 2008), and/or decreases the weight of negative (zero) samples during model training (Yu, Bilenko, and Lin 2017).
    
- ***meta learning : Learning to learn (fast)***
    
    > (KOR) [https://talkingaboutme.tistory.com/entry/DL-Meta-Learning-Learning-to-Learn-Fast](https://talkingaboutme.tistory.com/entry/DL-Meta-Learning-Learning-to-Learn-Fast)
    (ENG) [https://lilianweng.github.io/posts/2018-11-30-meta-learning/](https://lilianweng.github.io/posts/2018-11-30-meta-learning/)
    > 

Fixed weighting function은 전반적인 최적화 프로세스에서 사용되기 때문에 : cannot address the different characteristics of users or POIs (Zhang et al. 2018)

→ adaptive한 weighting scheme을 사용, 즉 data를 통해서 sample을 reweight 하는 방법을 배운다. 

But POI의 경우 비용 및 개인정보 문제로 인해 깨끗한 validation set을 얻기 힘들기 때문에, meta-data를 확보하기 어렵워 매우 challenging하다. 그러므로 본 논문에서는 meta-learning을 이용한 sample reweighting을 통해 meta data의 부족을 해결하고자 함.

- meta-data의 생성을 위해 : temporal ensembling (Laine and Aila 2017; Tarvainen and Valpola 2017)을 사용
- PREMERE :  recommender model + meta-model(i.e., weighting function)
![image](https://user-images.githubusercontent.com/63237947/174501976-b13462d9-6369-4783-b917-aac4a4e2cdc0.png)

1. 파라미터 $w$를 가진 추천 모델은 에폭마다 사용자가 each POI에 따라 방문할 확률을 예측
2. 만약 예측 결과가 충분한 수의 연속된 에폭에서 안정화되어 있다면, (temporal ensembling의 정의에 따라) 예측 결과가 Ground Truth와 같다고 간주한다. 
3. 2의 경우 예측된 결과로 user-POI Pair를 annotation하고,  이는 meta-model의 prarm $\theta$ 를 training하기 위한 meta-data로 사용할 수 있다. 

**Contributions**

- We propose PREMERE, a meta-learning approach to employ adaptive sample reweighting to mitigate the class imbalance and noisy values of the input data. Furthermore, a novel meta model architecture, PREMERE-NET, is developed to cope with the class imbalance issue. To the best of our knowledge, our work is the first attempt to adopt meta-learning for POI recommendation.
- We alleviate the absence of the meta-data by its self-generation via temporal ensembling, which is used to update the PREMERE-NET model that understands the characteristics of users and POIs.
- The recommendation accuracy improves by 2.36–26.9% for three real-world benchmark datasets when a heuristic weighting function is replaced with PREMERE in the state-of-the-art recommender algorithms. Note that PREMERE can be applied to any recommender algorithm.
